### DISCUSSION TITLE
Confused about IP addresses to use during setup

### DISCUSSION THREAD

**Discussion by iqworks** [Security-Onion-Solutions/securityonion#5063] [General]
Hi, my VMnets are setup like this:
(I am using VMware Workstation 16 pro)

[iq SO STENODO Bridged3.pdf](https://github.com/Security-Onion-Solutions/securityonion/files/6933648/iq.SO.STENODO.Bridged3.pdf)

VMnet0 - auto bridged.
VMnet4- host-only.
VMnet5 - custom - not used anywhere else.
VMnet8 - NAT.

here is what i put, but i dont think it is right?
manager IP - i used the IPv4 of my laptop from ipconfig.
access URL - is my IP from ipconfig
allowed IP - i was told to use a VMnet not used by anything else, so i put in 
                    VMnet5 so 192.168.50.17
and of course when i tryed my laptops IPv4 in https://192.168.x.x, i got access denied errors. 
so i am a little lost when it comes to what IPv4 i should use and where..

thanks for any advice or suggestions

**Comment by fluffyunicorn24** [comment]
Have you tried doing an so-allow with your IP from the SO console?

**Reply by iqworks** [reply]
Thanks for getting back. Yea, that’s the main issue, I am not getting to my SOC from my host chrome browser.
 I showed the IPs I used but I don’t think they are right. 
I tried https://192.168.50.17 and my host name and no go. Is there other information I can give you
For this? I am including another PDF with more information about VM setup.  it also shows the results of my so-allow.
[security sudo stuff.pdf](https://github.com/Security-Onion-Solutions/securityonion/files/6941329/security.sudo.stuff.pdf)
thanks again for your advice and suggestions

**Reply by fluffyunicorn24** [reply]
Try leaving off the subnet... like this:
so-allow
a
"the internal ip of your host system you want to access the gui from"

**Reply by iqworks** [reply]
Ok, not sure what you meant by "leaving off the subnet" ? 
 did sudo so-allow, a and my NAT IP and my NAT IP 192.168.80.0, which is on VMnet8. 
I tried https://192.168.80.5  in my host chrome, no go. 

I am using my host IPv4 in the first post I made, should I have used that? Or should I 
Make up an IP address, or how do I figure out how to do this?

I did a “sudo install_summary” its in this PDF (the missing parts are the last parts of my host ipconfig IPv4 address).
[security onion install logl.pdf](https://github.com/Security-Onion-Solutions/securityonion/files/6941952/security.onion.install.logl.pdf)
thanks as always

**Reply by fluffyunicorn24** [reply]
I might be wrong, but it sounds like you may have mis-configured your IPs.  I have VMWorkstation Pro at home I can try it on, but if you can, try installing an ubuntu desktop image and assign it the same VMnet your SO is on.  Then do a so-allow for that IP and see what happens.

**Comment by iqworks** [comment]
Yea, I believe my configuration is off to. Did you get a chance to look at the PDFs to see how I setup the VM? 

Maybe that could give you additional ideas. Thanks for your time in reviewing them.

 

   So, you mean use “securityonion-2.3.61-STENODOCKER.iso” , and use Linux/Ubuntu instead of Linux/CentOS 7 ?? 

   Or you mean use “ubuntu-20.04.2.0-desktop-amd64.iso”?

**Reply by fluffyunicorn24** [reply]
You need a second VM that’s behind that NAT. Ubuntu is easy and free so I would suggest that. Set it up as another VM and try it that way.

**Reply by iqworks** [reply]
Thanks so much for even taking the time. 
OK, here is my plan:
1 – create a new SO VM. 
       Use two adapters. One using VMnet8 (NAT) and the other using VMnet0 (auto-bridging). 
       Linux - ubuntu 64 bit? (I am thinking that ubuntu would give me a desktop over the SO?)
       During setup,  make my NIC ens33.
       The management IP should be 192.168.80.0/24.
       The gateway IP is 192.168.80.1.
       Access URL should be 192.168.80.0/24
       Allowed IP will be 192.168.80.0/24.  
       Ip a to get the IP address.
       And use this address when using so-allow a-analyist 

Thanks again

**Reply by fluffyunicorn24** [reply]
Your plan sounds good, just a couple of notes:
-the second VM is to allow you a gui to access your SO in.  This is because you are setting this up on VMWorkstation and not a server with a domain.
-The management IP should be static as in choose an IP in the VMnet subnet range so if your VMNet subnet IP is 192.168.80.0/24 then you could do 192.168.80.10 or something.  
- Get the gateway IP from your NAT settings.  I have seen the gateways be .2 so I would make sure it matches. 
-When it is done setting up, go to your Ubuntu system, open Firefox and type https://<ManagementIP>
-if you cant access it, check your IP on the Ubuntu and add it into the SO with a so-allow --> option a --> <IP of your Ubuntu>

**Comment by iqworks** [comment]
Thanks,  a second VM that’s behind that NAT? Not sure what you mean. 

Since I am pretty much new to this VM stuff, I am trying to get familiar with the vernacular.  

Do you mean to add another network card with another VMnet? Or something else. How do I 

Put a “second VM that’s behind that NAT” ?

 

What does “You need a second VM that’s behind that NAT” mean?

Does it mean I need to create a new VM using an Ubuntu iso?

**Reply by iqworks** [reply]
Hi, this is what is processing now.
[VMnew iq SO STENODO Bridged4 2.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6947242/VMnew.iq.SO.STENODO.Bridged4.2.txt)

**Comment by iqworks** [comment]
Ok, what does “you need a second VM that’s behind that NAT” mean? Are you saying I need to add another VMnet network to this VM network card?   

Or are you saying that I need to the security onion iso, but while I am installing it, I need to select linux/ubuntu or are you saying I need to 

Make a new VM from a new ubuntu iso with the same install and setup parameters I have used in the PDF I sent you to setup my security onion VM?

**Reply by fluffyunicorn24** [reply]
It’s ok! I still get thrown off when I use vmworkstation sometimes. You need your second vm to be on the same vnet as your SO (the one that is NAT). 

I apologize I’m a little tired now, so it will be easier for me to just explain what the typical steps are. I can try to look over your configs tomorrow. But so you don’t have to wait here’s how I set up mine:

Set up your Security Onion vm with 2 network adapters as well as the other hardware requirements. One network adapter should be NAT, the other can be bridged. If you need to see your VMnet settings go to Edit –> Virtual Network Editor –> Select the NAT network –> Select NAT Settings. 

When you go through the install of SO and it asks for your management NIC, ens33 and then the IP- put your NAT IP address here. Aka the IP in the VMNet you assigned as NAT. Again this can be found in Edit –> Virtual Network Editor –> Select the NAT network –> Select NAT Settings

When you finish setting up SO, configure the other VM, like Ubuntu and assign it to the same VMnet that is NAT and assigned to your SO. 

On the Ubuntu machine in terminal do a “ip a” and get the IP address. Make sure you do an so-allow for that IP. 

I hope this helps. 
>

**Comment by iqworks** [comment]
Thanks! I am doing it now.

 

From: fluffyunicorn24 ***@***.***> 
Sent: Friday, August 6, 2021 9:41 AM
To: Security-Onion-Solutions/securityonion
***@***.***>
Cc: iqworks ***@***.***>; Author ***@***.***>
Subject: Re: [Security-Onion-Solutions/securityonion] Confused about IP
addresses to use during setup (#5063)

 

Your plan sounds good, just a couple of notes:
-the second VM is to allow you a gui to access your SO in. This is because
you are setting this up on VMWorkstation and not a server with a domain.
-The management IP should be static as in choose an IP in the VMnet subnet
range so if your VMNet subnet IP is 192.168.80.0/24 then you could do
192.168.80.10 or something.

*	Get the gateway IP from your NAT settings. I have seen the gateways
be .2 so I would make sure it matches.
-When it is done setting up, go to your Ubuntu system, open Firefox and type
https://
-if you cant access it, check your IP on the Ubuntu and add it into the SO
with a so-allow --> option a --> 

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Security-Onion-Solutions/securityonion/discussions/5063#
discussioncomment-1140201> , or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ABKPV3YLO6HQTNETYKARLFTT3
QF75ANCNFSM5BRWI6GQ> .
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&m
t=8&pt=524675>  or Android
<https://play.google.com/store/apps/details?id=com.github.android&utm_campai
gn=notification-email> .
<https://github.com/notifications/beacon/ABKPV3YROU32PHYSBJCFHOLT3QF75A5CNFS
M5BRWI6G2YY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZG
OAAIWL2I.gif> 



-- 
This email has been checked for viruses by AVG.
https://www.avg.com

**Comment by iqworks** [comment]
I just checked that txt file I sent you and all of the step by step images
are missing? 

I will try another way tomorrow.

 

Now reading your instructions again, it looks like I need to create 2 VM one
SO with 2 VMnets, and,

An ubunto vm with the same 2 VMnets?

 

From: fluffyunicorn24 ***@***.***> 
Sent: Friday, August 6, 2021 9:41 AM
To: Security-Onion-Solutions/securityonion
***@***.***>
Cc: iqworks ***@***.***>; Author ***@***.***>
Subject: Re: [Security-Onion-Solutions/securityonion] Confused about IP
addresses to use during setup (#5063)

 

Your plan sounds good, just a couple of notes:
-the second VM is to allow you a gui to access your SO in. This is because
you are setting this up on VMWorkstation and not a server with a domain.
-The management IP should be static as in choose an IP in the VMnet subnet
range so if your VMNet subnet IP is 192.168.80.0/24 then you could do
192.168.80.10 or something.

*	Get the gateway IP from your NAT settings. I have seen the gateways
be .2 so I would make sure it matches.
-When it is done setting up, go to your Ubuntu system, open Firefox and type
https://
-if you cant access it, check your IP on the Ubuntu and add it into the SO
with a so-allow --> option a --> 

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Security-Onion-Solutions/securityonion/discussions/5063#
discussioncomment-1140201> , or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ABKPV3YLO6HQTNETYKARLFTT3
QF75ANCNFSM5BRWI6GQ> .
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&m
t=8&pt=524675>  or Android
<https://play.google.com/store/apps/details?id=com.github.android&utm_campai
gn=notification-email> .
<https://github.com/notifications/beacon/ABKPV3YROU32PHYSBJCFHOLT3QF75A5CNFS
M5BRWI6G2YY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZG
OAAIWL2I.gif> 



-- 
This email has been checked for viruses by AVG.
https://www.avg.com

**Reply by iqworks** [reply]
Hi fluffyunicorn24, sorry about sending you empty text. But I am going to send you two PDFs. 
i created a bridged CentOS VM and the Ubuntu VM.
But the IP addresses still look out of whack , but maybe its just me. 
thanks again for looking at these when you have time. 

[VMnew iq SO STENODO Bridged 64 5.pdf](https://github.com/Security-Onion-Solutions/securityonion/files/6948529/VMnew.iq.SO.STENODO.Bridged.64.5.pdf)
[VMnew iq Ubutnu for SO STENODO 64 5.pdf](https://github.com/Security-Onion-Solutions/securityonion/files/6948530/VMnew.iq.Ubutnu.for.SO.STENODO.64.5.pdf)

**Reply by fluffyunicorn24** [reply]
Hey, were you able to get your setup to work?

**Reply by iqworks** [reply]
Thanks for checking with me. I am in the process of gathering more info for you. 
 i am using securityonion-2.3.61-MSEARCH.iso now.
i did
Sudo so-allow 
Sudo so-test, here I get   
![iq SO 2 3 61-MSEARCH-2-2021-08-09-11-23-52](https://user-images.githubusercontent.com/5569263/128758555-6b4cc084-1fbd-49a2-afde-02a643304927.png)
so-allow
![iq SO 2 3 61-MSEARCH-2-2021-08-09-09-52-26](https://user-images.githubusercontent.com/5569263/128759287-4286f2a1-5c78-4010-b09f-02949e7af7c0.png)
  
![iq SO 2 3 61-MSEARCH-2-2021-08-09-12-02-38](https://user-images.githubusercontent.com/5569263/128760414-85c98c31-923a-46e1-bcdd-acaac39614a1.png)
![iq SO 2 3 61-MSEARCH-2-2021-08-09-12-01-54](https://user-images.githubusercontent.com/5569263/128760418-dc3dc11f-a30a-4c64-b6de-7cf37b389dfd.png)
![image](https://user-images.githubusercontent.com/5569263/128761030-21a0a8ac-f281-4d10-9f60-66d2edce984b.png)

this is where i am at. 
hey, where is my /opt folder to run :
sudo cat /opt/so/log/tcpreplay/init.log 

i am pulling up ubuntu right now and follow your instructions.

is there any information i can provide to you?
so weird, only SO VMs give me these issues  :-)
i am using 192.168.80.10 in so-allow analysist. i guess this is the IP i should try to run in ubuntu?
 
thanks very much for the help.

**Reply by iqworks** [reply]
forgot, so i have to download PCAP files myself? i get this when i run 
![image](https://user-images.githubusercontent.com/5569263/128767543-7a9f28d5-90a5-4473-99f2-56474046e69d.png)


i was told to run this as well:
![image](https://user-images.githubusercontent.com/5569263/128766041-86038f4a-8104-4b28-ac10-dc852c3eaac5.png)

**Reply by fluffyunicorn24** [reply]
What version of VMWorkstation are you using? 

Can you go to Edit –> Virtual Network Editor –> Select the NAT network –> Select NAT Settings and screenshot what the settings are in there?

**Reply by iqworks** [reply]
here is my virtula network editor
![image](https://user-images.githubusercontent.com/5569263/128927009-9b8dba14-b12f-4858-b032-9fd730b3c708.png)
![image](https://user-images.githubusercontent.com/5569263/128927166-5fc1f596-b60d-4a74-bbde-d0b7e90eda79.png)


 hope i am not giving you to much. 
 08/10/2021 – I have tried to narrow my questions down. 
 
![image](https://user-images.githubusercontent.com/5569263/128925896-f6c834fe-fef1-4eb1-9b68-93cb411b280b.png)

have no idea what to enter here. If I use my laptops windows 10 ipconfig IPv4, I think this causes issues.
![image](https://user-images.githubusercontent.com/5569263/128925956-232effc1-c274-42ed-af94-206ab14594bc.png)

I am thinking that standard is the best way to go. But airgap always seems to work. should I be using airgap? I am still researching this one. 
![image](https://user-images.githubusercontent.com/5569263/128926042-2c90d45c-9478-42bf-be2c-8acf1b0e31ba.png)

I am thinking that to speed up the setup, I don’t have to select anything here because I can install them later?
![image](https://user-images.githubusercontent.com/5569263/128926119-7d248d68-8024-43e0-be5e-e31b5e16cafc.png)

Still researching this one, but this factors in somewhere. I think this is what causes me to get the second IP prompt.
![image](https://user-images.githubusercontent.com/5569263/128926243-c8103bdc-f7c0-4fe0-9c1d-5cef38806d4f.png)

I think selecting for configure ntp servers, causes this last allow IP prompt comes up later, and, again, I have no clue WHAT to put in here. 
![image](https://user-images.githubusercontent.com/5569263/128926304-476fddc4-c6d2-4687-a7a1-00e17c4120b4.png)

when one of my setup failed I took a look at the sosetup.log and got this (to me this looks like an IP is incorrect):
Data failed to compile:
No matching sls found or registry in env base curl:
6 could not resolve host: raw.githubusercontent.com; 
Unknown error command failed with error 6
Could not pull signature key file, please ensure connectivity to https:// raw.githubusercontent.com
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.

thanks again for your time

**Reply by fluffyunicorn24** [reply]
Ok. In you virtual network editor click on VMNet 8 (NAT) and select “NAT Settings” under virtual information. Take a screenshot of that please.

**Reply by fluffyunicorn24** [reply]
Also I would do air gap on your setup.

**Reply by iqworks** [reply]
Here you go 
![image](https://user-images.githubusercontent.com/5569263/128939302-5a3bc18e-8f7c-4fb7-83fd-d935a80493a6.png)
 
yep, using airgap will let me install. I have been trying to install without using airgap.  I am going through the forum to see differences using airgap and not using it. when i dont use it i get even more ip confusion :-) 
it seems like airgap sets up a way to go out to the internet without you having to worry about the allow IP?

 thanks as usual.

**Reply by iqworks** [reply]
does mean i can only use IP addresses between the start and end IP?
![image](https://user-images.githubusercontent.com/5569263/128940348-94044d71-8c75-4140-99d8-214985abf4f9.png)
